### PR TITLE
Improved type checking

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed
Improved type checking, either by using `isinstance` or merging several `isinstance` checks.

#### Other comments
There are still some instances remaining. Also, I'm not sure if code of the type `if type(a) == type(b):` is better off as `if isinstance(a, type(b)):`?

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/plotting/experimental_lambdify.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def __call__(self, args):
                 return None
             return result.real
         except (ZeroDivisionError, OverflowError, TypeError) as e:
-            if isinstance(e, ZeroDivisionError) or isinstance(e, OverflowError):
+            if isinstance(e, (ZeroDivisionError, OverflowError)):
                 return None
```
Comment (oscarbenjamin) [inline sympy/plotting/experimental_lambdify.py:181]: This should probably be:
```python
except (ZeroDivisionError, OverflowError):
    return None
except TypeError:
    pass
```

**Comment by oscarbenjamin** [comment]
Is this finished?

**Comment by oscargus** [comment]
> Is this finished?

I'd say so. I have not changed absolutely every used of `type`, but most and also trying to limit the number of file changed per PR...

**Comment by oscargus** [comment]
> most

at least many...

**Pull request merged**